ocaml/tests/test_vm_helpers.ml

```diff
@@ -524,6 +528,28 @@ let test_group_hosts_netsriov_with_allocated () =
             "Test-failure: Unexpected number of sriov network in test"
   )
 
+let test_get_group_key_anti_affinity () =
+  let __context = T.make_test_database () in
+  let vm = T.make_vm ~__context () in
+  let group = T.make_vm_group ~__context ~placement:`anti_affinity () in
+  Db.VM.set_groups ~__context ~self:vm ~value:[group] ;
```

Comment:
Those are duplicated code, I think the original helper func `on_pool_of_anti_affinity` is a good one, just need to add the creating and setting for anti-affinity logic into it.